packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -179,7 +176,7 @@ class VisualStudio {
   /// not user-controllable, unlike the install location of Visual Studio
   /// itself.
   String get _vswherePath => _fileSystem.path.join(
-    _platform.environment['PROGRAMFILES(X86)'],
+    _platform.environment['PROGRAMFILES(X86)']!,
```

Comment:
I suspect someone could end up in a state where this isn't set, so we probably had a latent bug here. On the other hand, I expect Flutter would not be the first thing someone in that state would have issues with, so we can probably wait and see if it's worth doing a check+throw with a clear error message.

Reply:
Added a tool exit with an error message, and a unit test.